portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -55,24 +95,51 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       maxCommentsToDisplay = DEFAULT_MAX_COMMENTS;
     }
 
+    // Obtain the page ID of comment history to display.
+    String pageIdParam = request.getParameter("pageId");
+    totalPages = (int) Math.ceil(((double) totalComments) / maxCommentsToDisplay);
+    try {
+      currentPageId = Integer.parseInt(pageIdParam);
+      // If the numeric input of page ID is invalid, set to 1.
+      currentPageId = Math.max(1, currentPageId);
+    } catch (NumberFormatException e) {
+      if (pageIdParam.equals("first")) {
```

Comment:
Use a switch block for going through the cases for pageIdParam. For the "prev", you can do something like currentPageId = Math.max(currentPageId, currentPageId - 1) to make sure you aren't going to page 0

Reply:
Got it, thanks!